src/libNetwork/P2PComm.cpp

```diff
@@ -68,7 +67,36 @@ static void close_socket(int* cli_sock)
     }
 }
 
-P2PComm::P2PComm() {}
+P2PComm::P2PComm()
+{
+    auto func = [this]() -> void {
+        while (true)
+        {
+            this_thread::sleep_for(chrono::seconds(BROADCAST_INTERVAL));
+            lock(m_broadcastToRemovedMutex, m_broadcastHashesMutex);
+            lock_guard<mutex> g(m_broadcastToRemovedMutex, adopt_lock);
+            lock_guard<mutex> g2(m_broadcastHashesMutex, adopt_lock);
+
+            for (unsigned int i = 0; i < BROADCAST_INTERVAL; ++i)
+            {
+                auto it = m_broadcastToRemoved.find(time(nullptr)
+                                                    - BROADCAST_EXPIRY - i);
+
+                if (m_broadcastToRemoved.end() == it)
+                {
+                    continue;
+                }
+
+                auto it2 = m_broadcastHashes.find(it->second);
+                m_broadcastHashes.erase(m_broadcastHashes.begin(), it2);
```

Comment:
Although we already had a discussion, I'd like to put the comment here as a reminder: `m_broadcastHashes` stores the hashes by their alphabetical order other than insertion time, so the erase operation here may not work as expected (which should remove the hash entries since the beginning until `it2`)